crates/sui-bridge-indexer/src/worker.rs

```diff
@@ -57,30 +57,32 @@ impl BridgeWorker {
 
     // Process a transaction that has been identified as a bridge transaction.
     fn process_transaction(&self, tx: &CheckpointTransaction, checkpoint: u64, timestamp_ms: u64) {
-        if let Some(event) = &tx.events {
-            event.data.iter().for_each(|ev| {
+        if let Some(events) = &tx.events {
+            events.data.iter().for_each(|ev| {
                 if ev.type_.address == BRIDGE_ADDRESS {
                     let token_transfer = match ev.type_.name.as_str() {
                         "TokenDepositedEvent" => {
                             println!("Observed Sui Deposit");
-                            // todo: handle deserialization error
-                            let event: MoveTokenDepositedEvent =
+                            // TODO: handle deserialization error
+                            let move_event: MoveTokenDepositedEvent =
                                 bcs::from_bytes(&ev.contents).unwrap();
                             Some(TokenTransfer {
-                                chain_id: event.source_chain,
-                                nonce: event.seq_num,
+                                chain_id: move_event.source_chain,
+                                nonce: move_event.seq_num,
                                 block_height: checkpoint,
                                 timestamp_ms,
                                 txn_hash: tx.transaction.digest().inner().to_vec(),
+                                // should this value come from ev.sender?
+                                // sender_address: move_event.sender_address,
+                                sender_address: ev.sender.to_string(),
```

Comment:
@patrickkuo thoughts?

Reply:
ev.sender is wrong, this will be the transaction sender's address which will always be the bridge client's address. will need to get the sender address from the content of the event

Reply:
ah sorry, for deposited event this is correct. The sender might be different from the "real" bridge sender only in approved and claimed events.

Reply:
added a `sender_address` and a `txn_sender` to differentiate. The data is the same but it is "semantically" different